### PR TITLE
Cleanup ArchaiusModule

### PR DESCRIPTION
**Pull request by elandau**
Simplify bindings in ArchaiusModule and support an externally provided A
ppConfig implementation instead of allowing ArcahiusModule to be inherited.

### CODE REVIEW

**Comment by brharrington** [comment]
This approach will still require using `Module.override` if something needs to be injected to create the initial AppConfig object. For example, say I wanted to inject `DiscoveryClient` to lookup the set of platformservice instances.

File: archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java [hunk 1]
```diff
@@ -139,37 +138,39 @@ public void afterInjection(I injectee) {
...
+    
+    public ArchaiusModule() {
+        this(null);
+    }
+    
+    public ArchaiusModule(AppConfig appConfig) {
+        this.appConfig = appConfig;
+    }
+    
+    public static ArchaiusModule fromAppConfig(final AppConfig config) {
```
Comment (brharrington) [inline archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java]: What is the rationale having the public constructor that takes an `AppConfig` and this method?

File: archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java [hunk 2]
```diff
@@ -139,37 +138,39 @@ public void afterInjection(I injectee) {
...
         requestInjection(listener);
-        
         bindListener(Matchers.any(), listener);
+        
+        bind(ConfigMapper.class).to(DefaultConfigMapper.class).in(Scopes.SINGLETON);
     }
     
     @Provides
     @Singleton
-    protected ConfigMapper createConfigMapper() {
```
Comment (brharrington) [inline archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java]: I haven't needed to do it yet, but I'm curious, how often is it expected that the user will  need to customize the `ConfigMapper`?

**Comment by elandau** [comment]
I modified the Module to force the use of Modules.override in order to provide a different implementation of AppConfig.  This is actually a better use of Guice Modules since modules really shouldn't have any state and passing AppConfig to a module violates that principle.

File: archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java (hunk 2 above)
Comment (elandau) [inline archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java]: I don't expect a need for a different ConfigMapper implementation.

**Pull request merged**